docs/api-reference/retention-rules-api.md

```diff
@@ -25,45 +25,464 @@ sidebar_label: Retention rules
 
-## Retention rules
+Druid uses retention rules to determine what data is retained in the cluster. Druid supports load, drop, and broadcast rules. See [using rules to drop and retain data](../operations/rule-configuration.md) for more information. 
 
-Note that all _interval_ URL parameters are ISO 8601 strings delimited by a `_` instead of a `/` as in `2016-06-27_2016-06-28`.
+In this document, `http://SERVICE_IP:SERVICE_PORT` is a placeholder for the server address of deployment and the service port. For example, on the quickstart configuration, replace `http://ROUTER_IP:ROUTER_PORT` with `http://localhost:8888`.
 
-`GET /druid/coordinator/v1/rules`
+## Update retention rules for a datasource
 
-Returns all rules as JSON objects for all datasources in the cluster including the default datasource.
+Update one or more retention rules for a datasource. Retention rules can be submitted as an array of rule objects in the request body and overwrite any existing rules for the datasource. Rules are read in the order in which they appear, see [rule structure](../operations/rule-configuration.md) for more information.
 
-`GET /druid/coordinator/v1/rules/{dataSourceName}`
+Note that this endpoint returns an `HTTP 200 Success` code message even if the `datasource` does not exist.
 
-Returns all rules for a specified datasource.
+### URL
+<code class="postAPI">POST</code> `/druid/coordinator/v1/rules/{datasource}`
 
-`GET /druid/coordinator/v1/rules/{dataSourceName}?full`
+### Header parameters
 
-Returns all rules for a specified datasource and includes default datasource.
+The endpoint supports a set of optional header parameters to populate the `author` and `comment` fields in the `auditInfo` property for audit history. 
 
-`GET /druid/coordinator/v1/rules/history?interval=<interval>`
+* `X-Druid-Author` (optional)
+  * Type: String
+  * A string representing the author making the configuration change.
+* `X-Druid-Comment` (optional)
+  * Type: String
+  * A string describing the update.
 
-Returns audit history of rules for all datasources. Default value of interval can be specified by setting `druid.audit.manager.auditHistoryMillis` (1 week if not configured) in Coordinator `runtime.properties`.
+### Responses
 
-`GET /druid/coordinator/v1/rules/history?count=<n>`
+<!--DOCUSAURUS_CODE_TABS-->
 
-Returns last `n` entries of audit history of rules for all datasources.
+<!--200 SUCCESS-->
+<br/>
+*Successfully updated retention rules for specified datasource* 
 
-`GET /druid/coordinator/v1/rules/{dataSourceName}/history?interval=<interval>`
+<!--END_DOCUSAURUS_CODE_TABS-->
 
-Returns audit history of rules for a specified datasource. Default value of interval can be specified by setting `druid.audit.manager.auditHistoryMillis` (1 week if not configured) in Coordinator `runtime.properties`.
+---
+
+### Sample request
+
+The following example updates the retention rules for a datasource with specified ID `kttm1`.
```

Comment:
Do we actually call it an ID? I thought it was a name? Most of the IDs we talk about in Druid are of the autogenerated sort I think, so it'd be nice to have that consistency unless it is actually called ID.

Same q for all the other instnaces of this.